server/src/main/java/io/druid/metadata/SQLMetadataSupervisorManager.java

```diff
@@ -91,7 +92,7 @@ public Void withHandle(Handle handle) throws Exception
                     getSupervisorsTable()
                 )
             )
-                  .bind("id", String.format("%s_%s", id, version))
+                  .bind("id", String.format("%s_%s_%s", id, version, UUIDUtils.generateUuid()))
```

Comment:
The uuid should be part of the version too, right? Otherwise the `max(version)` thing for getting the most recent supervisor won't work.